astroid/interpreter/_import/spec.py

```diff
@@ -394,7 +424,7 @@ def find_spec(modpath: list[str], path: Sequence[str] | None = None) -> ModuleSp
             _path, modname, module_parts, processed, submodule_path or path
         )
         processed.append(modname)
-        if modpath:
+        if modpath and isinstance(finder, Finder):
```

Comment:
No tests fail when I revert this change.

Reply:
This is because of the typing change of `_find_spec_with_path`. `MetaPathFinder` is not a subclass of `Finder`, see:
https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder

Reply:
I'm still confused: won't this cause a behavior change across versions? On Python 3.9 this condition will match; on Py3.10 it won't.

Reply:
Yeah it might, it is just really hard to make this type safe..
`_MetaPathFinder` doesn't have this method so we can't safely call it.

We could use `hasattr` but that also feels unsatisfactory..

Reply:
Ah. I think we can't let type-checking considerations cause behavior differences.

Reply:
I'm dumb. `Finder != Finder`. The `Finder` in the `isinstance` check is actually our own implementation of `Finder`, which has `contribute_to_path`. `MetaPathFinders` don't so we can't call the method here.

Previously all `Finders` were of our own type but with this change this is no longer the case so we need to explicitly check this.